engine/launcher/src/main/scala/org/enso/launcher/Environment.scala

```diff
@@ -1,6 +1,159 @@
 package org.enso.launcher
 
+import java.io.File
+import java.nio.file.Path
+
+import scala.util.Try
+
+/**
+  * Gathers some helper methods querying the system environment.
+  *
+  * The default implementations should be used most of the time, but it is a
+  * trait so that the functions can be overridden in tests.
```

Comment:
As far as I understand, [objects cannot be subclassed in Scala](https://scala-lang.org/files/archive/spec/2.13/05-classes-and-objects.html#final).

Anyway, the classes that use this environment still need it as a parameter, because I need to be able to override the default one in tests, so this change wouldn't change much to how the dependencies are handled.

Do you have a suggestion on how else I could handle the fact that I need to override some implementations of the `Environment` in tests of other components that use the `Environment`?

Reply:
They can, you just need to extend `Object.type`, rather than `Object`.

Reply:
I did try that, but on this code:
```
object Test {
  object Base {
    def a: String = "a"
  }

  def tos(obj: Base.type): Unit = obj.a

  class Inherit extends Base.type {
    override def a: String = "b"
  }

  locally {
    tos(Base)
    tos(new Inherit)
  }

}
```

I'm getting the following error:
```
class type required but org.enso.launcher.components.Test.Base.type found
   class Inherit extends Base.type {
```

What am I doing wrong here?